### PR TITLE
refactor: Remove gmtime*

### PR DESCRIPTION
**Pull request by maflcko**
Now that the `ChronoSanityCheck` has passed for everyone with C++17 and is guaranteed by C++20 to always pass, remove it.

Also, remove `gmtime_r` and `gmtime_s` and replace them with `year_month_day`+`hh_mm_ss` from C++20.

### CODE REVIEW

**Comment by theuni** [comment]
Finally :)

Concept ACK.

**Comment by theuni** [comment]
> util/time.cpp:60:24: error: ‘year_month_day’ in namespace ‘std::chrono’ does not name a type

It would have been too easy if it had just worked in the real world :p

File: configure.ac [hunk 1]
```diff
@@ -1130,22 +1130,6 @@ if test "$use_thread_local" = "yes" || test "$use_thread_local" = "auto"; then
...
-  [ AC_MSG_RESULT([no]);
-    AC_MSG_CHECKING([for gmtime_s]);
-    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <ctime>]],
-       [[ gmtime_s((struct tm *) nullptr, (const time_t *) nullptr); ]])],
-       [ AC_MSG_RESULT([yes])],
-       [ AC_MSG_RESULT([no]); AC_MSG_ERROR([Both gmtime_r and gmtime_s are unavailable]) ]
-    )
-  ]
-)
-
```
Comment (fanquake) [inline configure.ac:1148]: Can also drop `AC_SUBST(HAVE_GMTIME_R)`.

**Review by fanquake — COMMENTED** [review]
Concept ACK - can we also drop the Boost date_time usage from `ParseISO8601DateTime` in favour of the std lib?

**Comment by fanquake** [comment]
> Probably no. std::chrono::parse isn't released yet in gcc at all, or did you mean something else?

Damn. I just meant any possibility to get rid of Boost.

**Comment by fanquake** [comment]
Want to rebase now that we've got 11.

**Comment by hebasto** [comment]
Concept ACK.

**Comment by fanquake** [comment]
cc @theuni you might want to circle back here now that this is unblocked.

**Review by fanquake — APPROVED** [review]
ACK fa9f36babaceba6ab2f88e64bc4bc2956f58871f - more std lib & even less stuff to port.

**Comment by theuni** [comment]
Post-merge utACK fa9f36babaceba6ab2f88e64bc4bc2956f58871f

**Pull request merged**